ci_framework/roles/env_op_images/tasks/main.yml

```diff
@@ -0,0 +1,111 @@
+          cifmw_edpm_prepare_extra_vars:
+
+    - name: Get images from the csv
+      ansible.builtin.shell: |
+        oc get csv -n {{ cifmw_install_yamls_defaults['OPERATOR_NAMESPACE'] }} openstack-operator.v0.0.1 -o=jsonpath='{.spec.install.spec.deployments[0].spec.template.spec.containers[1].env[*]}'
+      register: csv_images
+
+    - name: Format the output
+      ansible.builtin.set_fact:
+        csv_env_images: "{{ csv_images.stdout| regex_findall('\\{[^}]+\\}') }}"
+
+    - name: Extract env variable name and images
+      loop: "{{ csv_env_images }}"
+      ansible.builtin.set_fact:
+        csv_image_list: "{{ csv_image_list | default([]) + [item | from_json] }}"
+
+    - name: Write images to file
+      loop: "{{ csv_image_list }}"
+      ansible.builtin.lineinfile:
+        path: "{{ cifmw_env_op_images_dir }}/artifacts/{{ cifmw_env_op_images_file  }}"
+        line: "  {{item.name}}: {{item.value}}"
+        insertafter: "cifmw_edpm_prepare_extra_vars:"
+
+    - name: Get all the pods in openstack-operator namespace
+      kubernetes.core.k8s_info:
+        kind: Pod
+        namespace: "{{ cifmw_install_yamls_defaults['OPERATOR_NAMESPACE'] }}"
+        field_selectors:
+          - status.phase=Running
+      register: pod_list
+
+    - name: Retreive openstack-operator-index pod
+      ansible.builtin.set_fact:
+        selected_pod: "{{ pod_list.resources| selectattr('metadata.generateName', 'equalto', 'openstack-operator-index-') | list|first}}"
+
+    - name: Write openstack-operator-index image to file
+      ansible.builtin.lineinfile:
+        path: "{{ cifmw_env_op_images_dir }}/artifacts/{{ cifmw_env_op_images_file  }}"
+        line: "  OPENSTACK_IMG: {{selected_pod.status.containerStatuses[0].imageID}}"
+        insertafter: "cifmw_install_yamls_vars:"
+
+    - name: Retrieve rabbitmq-operator pod
+      ansible.builtin.set_fact:
+        selected_pod: "{{ pod_list.resources| selectattr('metadata.generateName', 'contains', 'rabbitmq-cluster-operator-') | list|first}}"
+
+    - name: Add rabbitmq operator image to the operator dictionary
+      ansible.builtin.set_fact:
+        operator_images_dict:
+          RABBITMQ_OP_IMG: "{{ selected_pod.status.containerStatuses[0].imageID}}"
+
+    - name: Retrieve rest of the operator pods
+      ansible.builtin.set_fact:
+        selected_pods: "{{ pod_list.resources| rejectattr('metadata.generateName', 'contains','openstack-operator-index-') | rejectattr('metadata.generateName', 'contains','rabbitmq-cluster-operator-') }}"
+
+    - name: Add operator images to the dictionary
+      loop: "{{ selected_pods }}"
+      ansible.builtin.set_fact:
+        operator_images_dict: "{{operator_images_dict | combine({ item.metadata.labels['openstack.org/operator-name'] | upper ~ '_OP_IMG': item.status.containerStatuses[1].imageID}) }}"
+
+    - name: Write openstack operator images to file
+      loop: "{{ operator_images_dict | dict2items }}"
+      ansible.builtin.lineinfile:
```

Comment:
Please, use `ansible.builtin.copy` and the `to_nice_yaml` filter to properly create a YAML file within Ansible.